### PR TITLE
Added Overlapping SG

### PR DESCRIPTION
**Pull request by jkabhishek**

### CODE REVIEW

File: plugins/ec2/overlappingSecurityGroup.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+                            permission.IpRanges[0].CidrIp + '#' +
+                            permission.FromPort + '#' +
+                            permission.ToPort;
+
+                        if (Object.keys(array_open_hash_map).includes(strhash)) {
+                            msg = group.GroupId + ' duplicated with ' + array_open_hash_map[strhash]
+                            helpers.addResult(results, 1,
+                                'Security group: ' + group.GroupId +
+                                ' and ' + array_open_hash_map[strhash] +
+                                'overlaps each other ', region, resource);
```
Comment (matthewdfuller) [inline plugins/ec2/overlappingSecurityGroup.js]: Can you add `found = true` here?

File: plugins/ec2/overlappingSecurityGroup.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+                        } else {
+                            array_open_hash_map[strhash] = group.GroupId;
+                        }
+
+                    }
+                }
+            }
```
Comment (matthewdfuller) [inline plugins/ec2/overlappingSecurityGroup.js]: Then check `if (!found) { // add a passing result}` here

**Comment by jkabhishek** [comment]
Updated Change Requested

File: plugins/ec2/overlappingSecurityGroup.js [hunk 3]
```diff
@@ -1,100 +1,61 @@
...
+                    ['ec2', 'describeSecurityGroups', region, sg.GroupId]);
+                        debugger;
 
-            var found = false;
-            var groups = describeSecurityGroups.data;
 
-            // loop through each  group
-            var array_open_hash_map = {};
-            for (group of groups) {
+                        // it is giving null here
```
Comment (jkabhishek) [inline plugins/ec2/overlappingSecurityGroup.js]: can you please look here

**Comment by matthewdfuller** [comment]
Closing in favor of https://github.com/cloudsploit/scans/pull/88

**Pull request closed without merging**